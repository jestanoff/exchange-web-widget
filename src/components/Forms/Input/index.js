import React from 'react';
import { oneOf, string } from 'prop-types';
import styles from './input.css';
import classNames from 'classnames';

const Input = ({ name }) => (
  <input
    className={classNames(styles.input, { [styles.from]: name === 'from' })}
    data-test={`input-${name}`}
    maxLength="17"
    name={name}
    placeholder="0"
    required
    type="tel"
  />
);

Input.propTypes = {
  name: oneOf(['from', 'to']).isRequired,
};

export default Input;
