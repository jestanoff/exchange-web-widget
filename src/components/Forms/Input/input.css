.input {
  background: transparent;
  border: none;
  color: #000000;
  font-size: 36px;
  text-align: right;
}

.input::placeholder {
  color: #BBBBBB;
}

.from {
  color: #7A838D;
}