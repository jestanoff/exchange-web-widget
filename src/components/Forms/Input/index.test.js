import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Input from './';


describe('Input component', () => {
  test('should render proper markup', () => {
    const { container } = render(<Input name="field" />);
  
    expect(container).toMatchSnapshot();
  })
});
