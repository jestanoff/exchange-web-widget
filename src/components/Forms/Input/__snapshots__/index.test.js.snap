// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render proper markup 1`] = `
<div>
  <input
    class="input"
    data-test="input-to"
    maxlength="17"
    name="to"
    placeholder="0"
    required=""
    type="tel"
  />
</div>
`;
